octobot_evaluators/evaluators/abstract_evaluator.py

```diff
@@ -86,6 +91,48 @@ def __init__(self, tentacles_setup_config: tm_configuration.TentaclesSetupConfig
         # Define evaluators default consumer priority level
         self.priority_level: int = channel_enums.ChannelConsumerPriorityLevels.MEDIUM.value
 
+        self.consumers = []
+
+        # True when this evaluator is only triggered on closed candles
+        self.is_triggered_after_candle_close = False
+
+        # Other evaluators that might have been called by this evaluator.
+        # This evaluator is now responsible for managing their cache
+        self.called_nested_evaluators = set()
+        if should_trigger_post_init:
+            self.post_init(tentacles_setup_config)
+
+    def post_init(self, tentacles_setup_config):
+        """
+        Automatically called after __init__ when post_init is True (default)
+        Override when necessary
+        :param tentacles_setup_config: the tentacles_setup_config __init__ argument
+        :return: None
+        """
+        pass
+
+    @classmethod
+    async def single_evaluation(
+            cls,
+            tentacles_setup_config: tm_configuration.TentaclesSetupConfiguration,
+            specific_config: dict,
+            ignore_cache=False,
+            **kwargs):
+        should_trigger_post_init = kwargs.pop("should_trigger_post_init", False)
```

Comment:
Why using a kwargs for this argument?

Reply:
I guess it made sense at some point, I removed it